### PR TITLE
Check that the node is not null, add contains to OpMap

### PR DESCRIPTION
**Pull request by larroy**
see title

### CODE REVIEW

File: nnvm/src/pass/gradient.cc [hunk 1]
```diff
@@ -143,23 +143,23 @@ Graph Gradient(Graph src) {
         << "because it is unreachable from the outputs.";
   }
 
-  // construct mirror reduece memory strategy if needed
+  // construct mirror as memory reduction strategy if needed
```
Comment (larroy) [inline nnvm/src/pass/gradient.cc]: is this mirror functionality used somewhere? at least I couldn't find usages in MXNet

File: nnvm/include/nnvm/op.h [hunk 2]
```diff
@@ -340,6 +340,13 @@ class OpMap {
    */
   inline int count(const Op* op) const;
 
+  /*!
+   * \brief Check if the map has op as key.
+   * \param op The key to the map
+   * \return true if op is contained in map, false otherwise.
+   */
+  inline bool contains(const Op* op) const;
```
Comment (tqchen) [inline nnvm/include/nnvm/op.h]: contains is just same as count so it is not necessary to add this function.

Comment (larroy) [reply nnvm/include/nnvm/op.h]: C++20 adds this function, I think is more readable, would be good to have it as well, don't you think is nice?

https://en.cppreference.com/w/cpp/container/unordered_map

Comment (larroy) [reply nnvm/include/nnvm/op.h]: @tqchen ping

File: nnvm/include/nnvm/op.h [hunk 3]
```diff
@@ -539,9 +546,18 @@ inline Op& Op::set_attr_parser(std::function<void (NodeAttrs* attrs)> fn) {  //
 // member functions of OpMap
 template<typename ValueType>
 inline int OpMap<ValueType>::count(const Op* op) const {
-  if (op == nullptr) return 0;
+  if (contains(op))
+    return 1;
```
Comment (tqchen) [inline nnvm/include/nnvm/op.h]: nit, always enclose by { } if the function contains multi-lines

File: nnvm/include/nnvm/op.h [hunk 4]
```diff
@@ -539,9 +546,18 @@ inline Op& Op::set_attr_parser(std::function<void (NodeAttrs* attrs)> fn) {  //
...
+  else
+    return 0;
+}
+
+template<typename ValueType>
+inline bool OpMap<ValueType>::contains(const Op* op) const {
+  if (op == nullptr)
+    return false;
```
Comment (tqchen) [inline nnvm/include/nnvm/op.h]: always enclose by {}

Comment (larroy) [reply nnvm/include/nnvm/op.h]: done

**Comment by tqchen** [comment]
Thanks, @larroy , this is merged

**Comment by larroy** [comment]
Thanks @tqchen

**Pull request merged**